ouroboros-consensus/test-consensus/Test/ThreadNet/RealPBFT.hs

```diff
@@ -194,9 +202,33 @@ tests = testGroup "RealPBFT" $
             , slotLengths  = defaultSlotLengths
             , initSeed     = seed
             }
+    , testProperty "Correct EpochNumber in delegation certificate" $
+      -- Caught a bug where the epoch of the delegation certificate that
+      -- culminates a rekey was being set according when the delegation
+      -- transaction was created instead of when it would mature (ie 2k slots
+      -- later).
+      let ncn4 = NumCoreNodes 4 in
+      prop_simple_real_pbft_convergence
+        NoEBBs
+        (SecurityParam 3)  -- epoches 0 30 60 90
+        TestConfig
+          { numCoreNodes = ncn4
+          , numSlots     = NumSlots 72
+          , nodeJoinPlan = trivialNodeJoinPlan ncn4
+          , nodeRestarts = NodeRestarts (Map.fromList [(SlotNo 59,Map.fromList [(CoreNodeId 3,NodeRekey)])])
+          , nodeTopology = meshNodeTopology ncn4
+          , slotLengths  = defaultSlotLengths
+          , initSeed     = Seed (17364222041321661634,8266509462575908621,10410472349244348261,9332246846568887555,6178891282750652496)
+          }
     , testProperty "simple convergence" $
           \produceEBBs ->
-          forAll (SecurityParam <$> elements [5, 10])
+          -- TODO k = 1 causes various errors
```

Comment:
TODO Fix this.

@edsko: Would you confirm that `k = 1` is a valid parameter assignment?

Reply:
Yes. Even `k = 0` in _principle_ ought to be, but the Byron ledger spec doesn't like it. I _think_ the implementation ought to be able to handle it, but not totally sure about that. But I have seen no problems with `k = 1`.

Reply:
I'm deferring this to Issue 1511 (let k = 1 in RealPBFT). That Issue discusses some of the concrete challenges.